### PR TITLE
Update of grass7 leftovers to grass8 or version generalization

### LINKED ISSUE #665: [Bug] r.diversity: Potential GRASS 8 compatibility issue
**Name of the addon**
r.diversity

**Describe the bug**
The `r.diversity` addon seams to add additional checks and builds a path to a config directory using an hardcoded "`GRASS7`" or "`.grass7`" string. In the imminent release of GRASS 8, this should lead to a compatibility issue.

**To Reproduce**
1. Open a GRASS 8 installation on a computer that never had a GRASS 7 installation (No `GRASS7` folders in `APPDATA` on Windows, or no `.grass7` folder in the user profile for others like Linux)
2. Install r.diversity addon (like `g.extension r.diversity`.
3. Try launching an example command, like `r.diversity input=ndvi_map prefix=diversity alpha=0.5` (from the manual page)
4. See if there is an error.

**Expected behavior**
`r.diversity` should install and run in GRASS 8 like it did on GRASS 7

**Screenshots**
See the following lines in the addon's source code:
https://github.com/OSGeo/grass-addons/blob/b7a36ebd8001fc77cdf11d8673e4dc7b8ec2fdef/src/raster/r.diversity/r.diversity.py#L86-L101

**System description (please complete the following information):**
I don't have a GRASS 8 only installation to confirm issue, so irrelevant (sorry).
 - Operating System: [e.g. Windows, Linux, ... incl. version]
 - GRASS GIS version [e.g. 7.8.1]

**Additional context**
I observed this potential issue when proofreading the raster addons recently.

_Originally posted by @echoix in https://github.com/OSGeo/grass-addons/issues/664#issuecomment-1008345218_

### PR DESCRIPTION
**Pull request by neteler**
- update of `.grass7` to `.grass8`
- update of startup script `grass7x` to versionless `grass`
- update URLs in HTML pages
  - svn --> GitHub
  - addons without full URL
  - core modules with generic "grass-stable" URL (will auto-fwd on server to "grass80" as soon as G8.0.0 has been released)

Fixes #665

### CODE REVIEW

**Comment by neteler** [comment]
@echoix: if possible, please take a look at this PR.

**Comment by echoix** [comment]
Since I was not involved long enough during the preparation of GRASS 8, I might need some context on what changed.
Does the core grass project effectively create .grass8 config folder? What about the GISBASE (in fact the command `grass70` --> `grass`)?

File: src/general/g.cloud/cloud_unpack.py [hunk 1]
```diff
@@ -44,8 +44,8 @@
 logging.debug("Unpacking in <%s/%s/%s>" % (dbase, location, mapset))
 f.write("Unpacking in <%s/%s/%s>" % (dbase, location, mapset))
 
-# read for grass70 executable the path to gisbase
-for line in open(which.which("grass70")).readlines():
+# read for grass executable the path to gisbase
+for line in open(which.which("grass")).readlines():
     if line.startswith('    gisbase = "'):
         gisbase = line.split("=")[-1].split('"')[1]
```
Comment (neteler) [inline src/general/g.cloud/cloud_unpack.py:50]: This might be outdated as we nowadays have:

`grass --config path --tmp-location epsg:4326 --exec g.gisenv get=GISBASE`

Maybe to be addressed in a separate PR.

**Comment by neteler** [comment]
> Since I was not involved long enough during the preparation of GRASS 8, I might need some context on what changed. 

Yes, of course.

> Does the core grass project effectively create .grass8 config folder?

Yes, that's correct. Also because the C-based addons are not binary compatible to those of GRASS GIS 7.x.

> What about the GISBASE (in fact the command `grass70` --> `grass`)?

In GRASS GIS 8 we go versionless with the startup script. To get GISBASE, one could run:

`grass --config path --tmp-location epsg:4326 --exec g.gisenv get=GISBASE`

**Comment by echoix** [comment]
I couldn't edit the PR, nor a fork of it, so I followed the flow of the github web UI, and you have a PR on your branch for g.cloud

File: src/general/g.cloud/cloud_unpack.py (hunk 1 above)
Comment (echoix) [reply src/general/g.cloud/cloud_unpack.py:50]: Ok, we should take note of it for another PR

File: src/raster/r.exdet/r.exdet.html [hunk 2]
```diff
@@ -100,7 +100,7 @@ <h2>CITATION</h2>
 
 <ul><li>van Breugel, P. (2016) r.exdet, a GRASS GIS addon for the
 quantification of novel uni- and multi-variate environments. URL:
-https://grass.osgeo.org/grass70/manuals/addons/r.exdet.html</li></ul>
+https://grass.osgeo.org/grass7/manuals/addons/r.exdet.html</li></ul>
```
Comment (echoix) [inline src/raster/r.exdet/r.exdet.html:103]: This still redirects to grass78, is that what is wanted? 
In the #664 PR I let it alone initially since it was a citation and thought that the reference was valid at that time.

File: src/raster/r.pops.spread/pops-core/CITATION.cff [hunk 3]
```diff
@@ -16,6 +16,6 @@ authors:
 title: r.spread.sod - SOD spread model implemented as GRASS GIS module
 version: 29074a1
 date-released: 2018-06-21
-url: https://grass.osgeo.org/grass74/manuals/addons/r.spread.sod.html
+url: https://grass.osgeo.org/grass7/manuals/addons/r.spread.sod.html
```
Comment (echoix) [inline src/raster/r.pops.spread/pops-core/CITATION.cff:19]: Since it is a citation suggestion, is this still valid?

File: src/raster/r.mblend/README.md [hunk 4]
```diff
@@ -3,7 +3,7 @@ r.mblend
 
 Draft implementation of the [Mblend algorithm](http://www.sciencedirect.com/science/article/pii/S0098300416300012) as an add-on for [GRASS 7](https://grass.osgeo.org/grass7/).
```
Comment (echoix) [inline src/raster/r.mblend/README.md:4]: Grass 7 here, and https://grass.osgeo.org/grass7/ redirects to https://grass.osgeo.org/learn/, so might as well lead to the homepage?
```suggestion
Draft implementation of the [Mblend algorithm](http://www.sciencedirect.com/science/article/pii/S0098300416300012) as an add-on for [GRASS](https://grass.osgeo.org). 
```

File: src/raster/r.mblend/README.md [hunk 5]
```diff
@@ -3,7 +3,7 @@ r.mblend
 
 Draft implementation of the [Mblend algorithm](http://www.sciencedirect.com/science/article/pii/S0098300416300012) as an add-on for [GRASS 7](https://grass.osgeo.org/grass7/). 
 
-Manual and detailed description is available at the [GRASS manual pages](https://grass.osgeo.org/grass74/manuals/addons/r.mblend.html).
+Manual and detailed description is available at the [GRASS manual pages](https://grass.osgeo.org/grass7/manuals/addons/r.mblend.html).
```
Comment (echoix) [inline src/raster/r.mblend/README.md:6]:
```suggestion
Manual and detailed description is available at the [GRASS manual pages](https://grass.osgeo.org/grass8/manuals/addons/r.mblend.html).
```

File: src/raster/r.suitability.regions/r.suitability.regions.html [hunk 6]
```diff
@@ -43,7 +43,7 @@ <h3>Option 2 - focal area</h3>
 neighborhood below. Using the maximum as statistic, the central cell 
 would be assigned a value of 5. Using the median, it would be assigned 
 the value 2. See <a 
-href="https://grass.osgeo.org/grass78/manuals/r.neighbors.html" 
+href="https://grass.osgeo.org/grass-stable/manuals/r.neighbors.html"
```
Comment (echoix) [inline src/raster/r.suitability.regions/r.suitability.regions.html:46]: If we use grass-stable in the full links to the GRASS core manual pages, why are we setting the URLs to grass8 in the addons manual pages? (And in this PR, some are set to relative URLs)

File: src/imagery/i.modis/i.modis.html [hunk 7]
```diff
@@ -339,7 +339,7 @@ <h2>AUTHOR</h2>
 Luca Delucchi, Initial version: Google Summer of Code 2011; subsequently updated by further authors
 
 <h2>SOURCE CODE</h2>
-<p>Available at: <a href="https://github.com/OSGeo/grass-addons/tree/master/grass7/imagery/i.modis">i.modis source code</a> (<a href="https://github.com/OSGeo/grass-addons/commits/master/grass7/imagery/i.modis">history</a>)</p>
+<p>Available at: <a href="https://github.com/OSGeo/grass-addons/tree/master/grass8/imagery/i.modis">i.modis source code</a> (<a href="https://github.com/OSGeo/grass-addons/commits/grass8/src/imagery/i.modis">history</a>)</p>
 
 <!--
 <p>
```
Comment (echoix) [inline src/imagery/i.modis/i.modis.html:345]: If this is a full HTML file since there isn't any real module, and is the homepage for the subfolder addon, why is the header added, with `<body bgcolor="white">` tag opened, but not closed at the end? What is the mechanism for keeping these "meta-manual" pages updated with the builder style?

**Review by echoix — COMMENTED** [review]
Added some comments

File: src/imagery/i.modis/i.modis.html (hunk 7 above)
Comment (neteler) [reply src/imagery/i.modis/i.modis.html:345]: All manual pages belonging to a GRASS module are generated/completed during compilation, hence header/footer are auto-generated.
Only the overview pages (e.g., [i.modis](https://grass.osgeo.org/grass8/manuals/addons/i.modis.html), [i.landsat](https://grass.osgeo.org/grass8/manuals/addons/i.landsat.html), ...) have - so far - to be fully crafted as they are not belonging to a real GRASS module.

A Makefile system tweak would be welcome to apply the same build mechanism as e.g. for [vectorintro.html](https://grass.osgeo.org/grass80/manuals/vectorintro.html) ([code](https://github.com/OSGeo/grass/blob/releasebranch_8_0/vector/vectorintro.html)).

File: src/raster/r.exdet/r.exdet.html (hunk 2 above)
Comment (neteler) [reply src/raster/r.exdet/r.exdet.html]: ok, restored in f6ac6f802258fb70cbd4fe35eedb3e833e9c7020

File: src/raster/r.futures/r.futures.calib/r.futures.calib.html [hunk 8]
```diff
@@ -12,7 +12,7 @@ <h2>DESCRIPTION</h2>
 
 <p>
 This module depends on addon
-<a href="https://grass.osgeo.org/grass7/manuals/addons/r.object.geometry.html">r.object.geometry</a>.
+<a href="r.object.geometry.html">r.object.geometry</a>.
```
[the remark this answers is not in the document]
Comment (neteler) [reply src/raster/r.futures/r.futures.calib/r.futures.calib.html:15]: I have updated the PR description: What we use is:

- addons without full URL
- core modules with generic "grass-stable" URL (will auto-fwd on server to "grass80" as soon as G8.0.0 has been released)

File: src/raster/r.mblend/README.md (hunk 4 above)
Comment (neteler) [reply src/raster/r.mblend/README.md]: Agreed.

File: src/raster/r.mblend/README.md (hunk 5 above)
Comment (neteler) [reply src/raster/r.mblend/README.md]: Addressed in 59513d57aadcdc43a97291fcadd06ffc9b55fd62

**Comment by echoix** [comment]
I saw in one of these files a mention of the test using the North Carolina data sets, and I see it often in the tests, I assume it is always present in the testing environment. See:

> When your are writing tests you can rely on having maps which are present in the NC sample location, or you can generate random maps.

(From https://grass.osgeo.org/grass80/manuals/libpython/gunittest_testing.html#test-general)

However, I'm working on a issue/PR and analyzing the CI output, and I would need a precision on which dataset is used exactly. Since there are 31 files in https://grass.osgeo.org/sampledata/north_carolina/, which is used? The CI uses https://grass.osgeo.org/sampledata/north_carolina/nc_spm_full_v2alpha2.tar.gz, but some failing tests use "basins" that I found in https://grass.osgeo.org/sampledata/north_carolina/nc_basic_spm_grass7.tar.gz.

File: src/raster/r.pops.spread/pops-core/CITATION.cff (hunk 3 above)
Comment (neteler) [reply src/raster/r.pops.spread/pops-core/CITATION.cff:19]: It seems that page got renamed? I cannot find "r.spread.sod.html"...

File: src/raster/r.mblend/README.md [hunk 9]
```diff
@@ -1,9 +1,9 @@
 r.mblend
 ==========
 
-Draft implementation of the [Mblend algorithm](http://www.sciencedirect.com/science/article/pii/S0098300416300012) as an add-on for [GRASS 7](https://grass.osgeo.org/grass7/). 
+Draft implementation of the [Mblend algorithm](http://www.sciencedirect.com/science/article/pii/S0098300416300012) as an add-on for [GRASS 7](https://grass.osgeo.org/).
```
Comment (echoix) [inline src/raster/r.mblend/README.md:4]:
```suggestion
Draft implementation of the [Mblend algorithm](http://www.sciencedirect.com/science/article/pii/S0098300416300012) as an add-on for [GRASS](https://grass.osgeo.org).
```

**Review by echoix — CHANGES_REQUESTED** [review]

File: src/raster/r.suitability.regions/r.suitability.regions.html (hunk 6 above)
Comment (neteler) [reply src/raster/r.suitability.regions/r.suitability.regions.html:46]: The best way to avoid hardcoded versions is using the "grass-stable" redirect on the server: modules with generic "grass-stable" URL currently still point to G78 but will be auto-fwd'ed on server to "grass80" as soon as G8.0.0 has been released.

File: src/gui/wxpython/wx.wms/readme.txt [hunk 10]
```diff
@@ -15,13 +15,13 @@ Libraries to be installed
 
 How to run the add-on
 
-1) Create a directory wms in as grass_trunk/gui/wxpython/gui_modules/wms. Place all the files in the svn root/grass-addons/grass7/gui/wxpython/wx.wmswms directory in the created wms directory. 
+1) Create a directory wms in as grass_trunk/gui/wxpython/gui_modules/wms. Place all the files in the grass-addons/src/gui/wxpython/wx.wms directory in the created wms directory. 
 2) Place the config file and ServersList.xml file in grass_trunk
 3) Patch wxgui.py and toolbars.py with command "cd path/to/sourceroot && cat core.diff | patch -p0"
-4) Patch grass_trunk/gui/wxpython/Makefile with comamnd "cd path/to/sourceroot && cat Makefile.diff | patch -p0"
-5) Place __init__.py file in the grass_trunk/gui/wxpython/gui_modules/wms . 
+4) Patch GRASS GIS core gui/wxpython/Makefile with comamnd "cd path/to/sourceroot && cat Makefile.diff | patch -p0"
```
Comment (nilason) [inline src/gui/wxpython/wx.wms/readme.txt:21]: Typo ”comamnd”

File: src/general/g.cloud/cloud_unpack.py (hunk 1 above)
Comment (neteler) [reply src/general/g.cloud/cloud_unpack.py:50]: For now a new issue: #668

Comment (neteler) [reply src/general/g.cloud/cloud_unpack.py:50]: > I couldn't edit the PR, nor a fork of it, so I followed the flow of the github web UI, and you have a PR on your branch for g.cloud

Thanks, merged into this PR.

File: src/raster/r.futures/r.futures.calib/r.futures.calib.html [hunk 11]
```diff
@@ -94,7 +94,7 @@ <h3>Calibration input and output</h3>
 
 <h2>NOTES</h2>
 This module depends on addon
-<a href="https://grass.osgeo.org/grass7/manuals/addons/r.object.geometry.html">r.object.geometry</a>.
+<a href="r.object.geometry.html">r.object.geometry</a>.
```
[the remark this answers is not in the document]
Comment (neteler) [reply src/raster/r.futures/r.futures.calib/r.futures.calib.html:97]: (see above)

File: src/raster/r.mblend/README.md (hunk 9 above)
Comment (neteler) [reply src/raster/r.mblend/README.md]: Fixed.

File: src/gui/wxpython/wx.wms/readme.txt (hunk 10 above)
Comment (neteler) [reply src/gui/wxpython/wx.wms/readme.txt:21]: fixed.

File: src/raster/r.pops.spread/pops-core/CITATION.cff (hunk 3 above)
Comment (neteler) [reply src/raster/r.pops.spread/pops-core/CITATION.cff]: Fixed in dbc78fa2c018604bd24a5e8c750aa6a8bc75b17f

**Comment by wenzeslaus** [comment]
> ...In GRASS GIS 8 we go versionless with the startup script. To get GISBASE, one could run:
> 
> `grass --config path --tmp-location epsg:4326 --exec g.gisenv get=GISBASE`

Isn't it either (not preferred!):

```
grass --tmp-location XY --exec g.gisenv get=GISBASE
```

or simply (preferred):

```
grass --config path
```

?

**Comment by neteler** [comment]
> However, I'm working on a issue/PR and analyzing the CI output, and I would need a precision on which dataset is used exactly. Since there are 31 files in https://grass.osgeo.org/sampledata/north_carolina/, which is used?

... I have moved this out to a separate issue (#674)
